mmf/common/batch_collator.py

```diff
@@ -21,11 +21,30 @@ def __call__(self, batch):
         ):
             sample_list = batch[0]
         elif not isinstance(batch, SampleList):
-            sample_list = SampleList(batch)
+            sample_list = _build_sample_list_with_detr_fields(batch)
 
         if sample_list._get_tensor_field() is None:
             sample_list = SampleList(sample_list.to_dict())
 
         sample_list.dataset_name = self._dataset_name
         sample_list.dataset_type = self._dataset_type
         return sample_list
+
+
+def _build_sample_list_with_detr_fields(batch):
```

Comment:
Thanks for the comments. I'll handle it.

Reply:
(this is now resolved; the batch collator stays the same)

Also, after a discussion with Amanpreet, we decided not to support dynamic image sizes in batching, since it is not used in our UniT model or any vision SSL model, while adding a large maintenance overhead.